gdal_subsetter/transform.py

```diff
@@ -311,11 +311,9 @@ def update_layernames(self, file_name, layer_names):
         layernames : string[]
             An array of names, in order, to apply to the layers
         """
-        dataset = gdal.Open(file_name)
-        for index, layer_name in enumerate(layer_names):
-            dataset.GetRasterBand(index + 1).SetDescription(layer_name)
-
-        dataset = None
+        with OpenGDAL(file_name) as dataset:
+            for index, layer_name in enumerate(layer_names, start=1):
+                dataset.GetRasterBand(index).SetDescription(layer_name)
```

Comment:
There was an `enumerate` statement that was recently removed (because the index wasn't used). But it had the `start=1` kwarg, which is super useful for these loops where GDAL band indices are 1-based.